examples/hooks-next/package.json

```diff
@@ -0,0 +1,30 @@
+{
+  "name": "hooks-next-example",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "algoliasearch": "^4.11.0",
+    "instantsearch.css": "^7.4.5",
+    "next": "12.0.7",
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
+    "react-instantsearch-hooks": "^6.17.0",
+    "react-instantsearch-hooks-server": "^6.17.0"
+  },
+  "resolutions": {
```

Comment:
I haven't been able to run the example locally or in CSB with the resolutions of those packages, but the code looks good to me. I'll retry when those packages will be released.